common/beerocks/bcl/source/network/network_utils.cpp

```diff
@@ -363,6 +365,102 @@ static int parseRoutes(struct nlmsghdr *nlHdr, std::shared_ptr<route_info> rtInf
     return (0);
 }
 
+std::shared_ptr<std::unordered_map<std::string, std::string>>
+network_utils::get_arp_table(bool mac_as_key)
+{
+    auto arp_table_ptr = std::make_shared<std::unordered_map<std::string, std::string>>();
+    auto &arp_table    = *arp_table_ptr.get();
+
+    // Allocate a new netlink socket
+    std::unique_ptr<nl_sock, decltype(&nl_socket_free)> nl_socket(nl_socket_alloc(),
+                                                                  &nl_socket_free);
+    if (!nl_socket) {
+        LOG(ERROR) << "Failed allocating netlink socket!";
+        return nullptr;
+    }
+
+    int err = nl_connect(nl_socket.get(), NETLINK_ROUTE);
+    if (err != 0) {
+        LOG(ERROR) << "Failed connecting the netlink socket: " << nl_geterror(err);
+        return nullptr;
+    }
+
+    // ARP table cache
+    struct nl_cache *neighbor_table_cache = nullptr;
+
+    // Allocate the cache and fill it with data
+    err = rtnl_neigh_alloc_cache(nl_socket.get(), &neighbor_table_cache);
```

Comment:
```suggestion
    struct nl_cache* tmp = nullptr;
    err = rtnl_neigh_alloc_cache(nl_socket.get(), &tmp);
```
[this example explains](https://stackoverflow.com/questions/12403750/initializing-a-stdunique-ptr-by-passing-the-address-of-the-pointer)

Reply:
If it could be done without declaring a "tmp" pointer I would do it.